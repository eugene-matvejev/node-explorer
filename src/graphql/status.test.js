describe('GraphQL Type: Status', () => {
    describe('query', () => {
        it(`fetch all statuses' only scalar values`, async () => {
            const { data } = await query({
                query: `
                {
                    statuses {
                        id
                        name
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`fetch only scalar values`, async () => {
            const { data } = await query({
                query: `
                {
                    status(id: 1) {
                        id
                        name
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`fetch parent's scalar values`, async () => {
            const { data } = await query({
                query: `
                {
                    status(id: 2) {
                        parent {
                            id
                            name
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        describe('search by pattern', () => {
            it(`by pattern only`, async () => {
                const { data } = await query({
                    query: `
                    {
                        searchStatus(pattern: "children") {
                            name
                        }
                    }`
                });

                expect(data).toMatchSnapshot();
            });

            it(`by pattern and enforced perPage/page variables`, async () => {
                const { data } = await query({
                    query: `
                    {
                        searchStatus(pattern: "children", perPage: 1, page: 1) {
                            name
                        }
                    }`
                });

                expect(data).toMatchSnapshot();
            });
        });
    });

    describe('mutation', () => {
        it(`should create new status, as status with same name do NOT exists yet`, async () => {
            const { data } = await query({
                query: `
                mutation {
                    addStatus(
                        input: {
                            name: "root"
                        }
                    ) {
                        id
                        name
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`should NOT create new status, as status with silimar name already exists`, async () => {
            const { errors } = await query({
                query: `
                mutation {
                    addStatus(
                        input: {
                            name: "status which exists"
                        }
                    ) {
                        id
                        name
                    }
                }`
            });

            expect(errors).toHaveLength(1);
        });


        it(`should update status, as such name do NOT exists yet`, async () => {
            const { data } = await query({
                query: `
                mutation {
                    updateStatus(
                        input: {
                            id: 1
                            name: "update status 1 name"
                        }
                    ) {
                        id
                        name
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });


        it(`should NOT update status name, as status with silimar name already exists`, async () => {
            const { errors } = await query({
                query: `
                mutation {
                    addStatus(
                        input: {
                            id: 2
                            name: "status 1"
                        }
                    ) {
                        id
                        name
                    }
                }`
            });

            expect(errors).toHaveLength(1);
        });
    });
});
