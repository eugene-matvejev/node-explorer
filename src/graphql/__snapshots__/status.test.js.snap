// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL Type: Status mutation should create new status, as status with same name do NOT exists yet 1`] = `
Object {
  "addStatus": Object {
    "id": "11",
    "name": "root",
  },
}
`;

exports[`GraphQL Type: Status mutation should update status, as such name do NOT exists yet 1`] = `
Object {
  "updateStatus": Object {
    "id": "1",
    "name": "update status 1 name",
  },
}
`;

exports[`GraphQL Type: Status query fetch all statuses' only scalar values 1`] = `
Object {
  "statuses": Array [
    Object {
      "id": "1",
      "name": "status 1",
    },
    Object {
      "id": "2",
      "name": "status 2",
    },
    Object {
      "id": "3",
      "name": "status 3",
    },
    Object {
      "id": "4",
      "name": "status which exists",
    },
    Object {
      "id": "5",
      "name": "status 1 children: 1",
    },
    Object {
      "id": "6",
      "name": "status 2 children: 1",
    },
    Object {
      "id": "7",
      "name": "status 2 children: 2",
    },
    Object {
      "id": "8",
      "name": "status 3 children: 1",
    },
    Object {
      "id": "9",
      "name": "status 3 children: 2",
    },
    Object {
      "id": "10",
      "name": "status 3 children: 3",
    },
  ],
}
`;

exports[`GraphQL Type: Status query fetch only scalar values 1`] = `
Object {
  "status": Object {
    "id": "1",
    "name": "status 1",
  },
}
`;

exports[`GraphQL Type: Status query fetch parent's scalar values 1`] = `
Object {
  "status": Object {
    "parent": Object {
      "id": "1",
      "name": "status 1",
    },
  },
}
`;

exports[`GraphQL Type: Status query search by pattern by pattern and enforced perPage/page variables 1`] = `
Object {
  "searchStatus": Array [
    Object {
      "name": "status 1 children: 1",
    },
  ],
}
`;

exports[`GraphQL Type: Status query search by pattern by pattern only 1`] = `
Object {
  "searchStatus": Array [
    Object {
      "name": "status 1 children: 1",
    },
    Object {
      "name": "status 2 children: 1",
    },
    Object {
      "name": "status 2 children: 2",
    },
    Object {
      "name": "status 3 children: 1",
    },
    Object {
      "name": "status 3 children: 2",
    },
    Object {
      "name": "status 3 children: 3",
    },
  ],
}
`;
